core/trino-main/src/main/java/io/trino/operator/output/SkewedPartitionRebalancer.java

```diff
@@ -153,11 +160,26 @@ public static SkewedPartitionRebalancer createSkewedPartitionRebalancer(
             int taskCount,
             int taskPartitionedWriterCount,
             long minPartitionDataProcessedRebalanceThreshold,
-            long maxDataProcessedRebalanceThreshold)
+            long maxDataProcessedRebalanceThreshold,
+            int maxPartitionsToRebalance)
     {
         // Keep the task bucket count to 50% of total local writers
         int taskBucketCount = (int) ceil(0.5 * taskPartitionedWriterCount);
-        return new SkewedPartitionRebalancer(partitionCount, taskCount, taskBucketCount, minPartitionDataProcessedRebalanceThreshold, maxDataProcessedRebalanceThreshold);
+        return new SkewedPartitionRebalancer(
+                partitionCount,
+                taskCount,
+                taskBucketCount,
+                minPartitionDataProcessedRebalanceThreshold,
+                maxDataProcessedRebalanceThreshold,
+                // Keep the maxPartitionsToRebalance to atleast task count such that single partition writes do
+                // not suffer from skewness and can scale uniformly across all tasks.
+                max(maxPartitionsToRebalance, taskCount));
+    }
+
+    public static int getScaleWritersMaxSkewedPartitions(Session session)
+    {
+        // Set the value of maxSkewedPartitions to scale to 60% of maximum number of writers possible per node.
+        return (int) (ceil((double) getQueryMaxMemoryPerNode(session).toBytes() / MAX_MEMORY_PER_PARTITION_WRITER) * 0.60);
```

Comment:
nit: consider adding `MAX_MEMORY_PER_PARTITION_WRITER` as a session property/feature config. Can be separate toggle.